container/docker/factory.go

```diff
@@ -232,18 +265,26 @@ func Register(factory info.MachineInfoFactory, fsInfo fs.FsInfo) error {
 		}
 	}
 
-	// Check that the libcontainer execdriver is used.
-	information, err := DockerInfo()
+	information, err := client.Info()
 	if err != nil {
 		return fmt.Errorf("failed to detect Docker info: %v", err)
 	}
-	execDriver, ok := information["ExecutionDriver"]
-	if !ok || !strings.HasPrefix(execDriver, "native") {
+
+	// Check that the libcontainer execdriver is used.
+	execDriver := information.Get("ExecutionDriver")
+	if execDriver == "" || !strings.HasPrefix(execDriver, "native") {
```

Comment:
nit: is it necessary to check if execDriver == "", because that doesn't have the prefix "native"?